### PR TITLE
Delete old nightlies

### LINKED ISSUE #5: Job to delete older binaries from anaconda.org
The nightly binaries use up a lot of storage space on anaconda.org. The plan is for me to write a GitHub Actions workflow that will run every Sunday and delete all nightly binaries that are more than a week old.

cc: @ihnorton

### PR DESCRIPTION
**Pull request by jdblischak**
This GitHub Actions pipelines deletes old nightlies. It is set to run every Monday and remove all nightlies that were uploaded more than 7 days ago

Closes #5 

cc: @ihnorton @Shelnutt2

## Setup steps

The following setup steps are required:

1. Go to https://anaconda.org/tiledb/settings/access and create a new API token with scope "Allow all API operations"
1. Create a [new repository secret](https://github.com/TileDB-Inc/conda-forge-nightly-controller/settings/secrets/actions) named `ANACONDA_TOKEN` and copy-paste the token

## Execution

The script can be run locally, eg:

```sh
mamba env create --file scripts/delete-old-nightlies.yml
mamba activate delete-old-nightlies
export ANACONDA_TOKEN="<api-token>"
ACCOUNT=tiledb DAYS=7 SUBDIR=osx-arm64 scripts/delete-old-nightlies.sh
```

Or executed remotely using the `workflow_dispatch` trigger:

image

## Example

On my fork, I used the `workflow_dispatch` trigger to delete all the nightlies from my account that were uploaded more than 140 days ago.

```sh
date --date="140 days ago"
## Tue Nov  1 15:03:38 EDT 2022
```

As can be seen in the [build log](https://github.com/jdblischak/conda-forge-nightly-controller/actions/runs/4482850391/jobs/7881415269#step:5:287), all the nightlies from October were deleted. And the only nightlies remaining in my account are from [November](https://anaconda.org/jdblischak/tiledb/files) (note that in a previous run I deleted all the linux-64 and osx-64 nightlies, which is why their November nightlies aren't there; also, for context, I stopped uploading nightlies to my channel in November when this repo was created)

### CODE REVIEW

**Review by ihnorton — APPROVED** [review]
Nice work! This LGTM with the one minor caveat that I think we should consider prefixing the variables with something like `TILEDB_CFNC_` to avoid any small risk of being overridden by something in the actions environment (this comes from, uh, [painful experience](https://github.com/TileDB-Inc/TileDB/blob/b082ee068c560e7d79feac366b164154ab1c7c86/scripts/azure-linux_mac-release.yml#L33-L39)).

**Comment by ihnorton** [comment]
> The following setup steps are required:

Done

Also tagging @eddelbuettel who has some interest and experience in automation like this for a second set of eyes.

**Review by eddelbuettel — APPROVED** [review]
Looks good to me but then ... I didn't know we left knightlies behind so maybe get approval from a second set of eyes before merging.

Only other added value I can offer is that I once spotted a hint in the GHA docs that their scheduled times are a little uneven across the hour.  So I tend to put mine at minute 17 or 39 or ...  Gotta make sure its prime :stuck_out_tongue_winking_eye:

**Comment by jdblischak** [comment]
> we should consider prefixing the variables with something like `TILEDB_CFNC_` to avoid any small risk of being overridden by something in the actions environment (this comes from, uh, painful experience).

@ihnorton That is nightmare fuel indeed. I just pushed a commit to prefix the env vars. I used `TILEDB_CI_` to avoid the (likely) possibility of me misspelling `CFNC`

**Comment by jdblischak** [comment]
> I once spotted a hint in the GHA docs that their scheduled times are a little uneven across the hour

@eddelbuettel From experience I can confirm that the CRON expressions are treated simply as suggestions by GitHub Actions. This pipeline only needs to run once a week to free up storage space on anaconda.org, so it's not a big deal if its start time is off

**Comment by jdblischak** [comment]
@ihnorton I don't have write access to this repo. Please merge after you add `ANACONDA_TOKEN` as a repository secret

**Comment by jdblischak** [comment]
Here's the build log from my fork of a dry run to confirm the updated env var names are consistent

https://github.com/jdblischak/conda-forge-nightly-controller/actions/runs/4514326213/jobs/7950272701#step:5:1

**Comment by eddelbuettel** [comment]
@jdblischak I meant that most as a "voluntary user-side load balancing help" to move away from the peak at the full hour.

**Comment by jdblischak** [comment]
> I meant that most as a "voluntary user-side load balancing help" to move away from the peak at the full hour.

Ah, ok. Now I understand. And I found it in the [docs](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#schedule):

> Note: The `schedule` event can be delayed during periods of high loads of
> GitHub Actions workflow runs. High load times include the start of every hour.
> To decrease the chance of delay, schedule your workflow to run at a different
> time of the hour.

I let R decide :-)

```R
> sample(5:55, 1)
[1] 36
```

**Comment by eddelbuettel** [comment]
The meta problem is of course how on earth you picked the seed </ducks>

**Comment by ihnorton** [comment]
@jdblischak I added you as a maintainer.

**Comment by jdblischak** [comment]
I was able to manually trigger the workflow:

https://github.com/TileDB-Inc/conda-forge-nightly-controller/actions/runs/4533066313/jobs/7985366120

The API token worked. I set `TILEDB_CI_DAYS=130`, and it removed all the nightlies prior to Nov 17, 2022

```sh
date --date="130 days ago"
## Thu Nov 17 09:30:22 EST 2022
```

https://anaconda.org/tiledb/tiledb/files
https://anaconda.org/tiledb/tiledb/files?version=2.14.0.2022_11_17

Next Monday, when the schedule workflow is triggered, it will delete hundreds more nightlies (`TILEDB_CI_DAYS=7`)

**Comment by ihnorton** [comment]
Great, thank you very much!!

**Comment by jdblischak** [comment]
Quick follow up: the first weekly scheduled job successfully deleted hundreds of older nightlies

https://github.com/TileDB-Inc/conda-forge-nightly-controller/actions/runs/4591764429/jobs/8108210275#step:5:17
https://anaconda.org/tiledb/tiledb/files?channel=nightlies
https://anaconda.org/tiledb/tiledb/files?version=2.16.0.2023_03_26&channel=nightlies

Hopefully this significantly increased our available storage space on anaconda.org

**Pull request merged**
